include/unifex/task.hpp

```diff
@@ -216,7 +217,7 @@ struct _promise {
       } else {
         return unifex::await_transform(
             *this,
-            typed_via(as_sender((Awaitable&&) awaitable), this->sched_));
+            finally(as_sender((Awaitable&&) awaitable), unstoppable(schedule(this->sched_))));
```

Comment:
Rather than make this change here, I wonder if it would be better to make this change in `typed_via()` so that it applies more widely.

`typed_via()` is trying to move execution onto some other context (like with `task`'s scheduler-affinity).
In many cases, it's a correctness issue if you don't end up continuing on that context.
e.g. maybe the context is a strand and you're relying on destructors running on that context to avoid certain data-races

So it may be worthwhile just saying that the schedule-op that typed_via() incorporates must complete with set_value(), and applying the unstoppable() to the schedule() for all usages of typed_via().

Reply:
Ah, I see.

I was being conservative, but yeah, I think you're right.

Reply:
This reliably deadlocks the `examples/linux/io_epoll_test` program so I'm going to revert to the original strategy.  Once the internal crashing bug is really put to bed, we can investigate whether the example program is just buggy.